### PR TITLE
Start ci workflow

### PR DESCRIPTION
**Pull request by sambarnes**
just passing some time in traffic so took a stab

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+18.9.0
```
Comment (sambarnes) [inline .nvmrc:1]: not much into the ts ecosystem personally, so open to suggestions here

**Comment by sambarnes** [comment]
using `act` to test locally

dunno why it didn't run here tho, maybe [some setting](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository) about the org/perms?

---

linting jobs:

```shell
$ act --container-architecture linux/amd64 --job linting
...
[CI/Linting]   ✅  Success - actions/setup-node@v3
[CI/Linting] ⭐  Run Install yarn
[CI/Linting]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/1-composite-1.sh] user= workdir=
| ...
| found 0 vulnerabilities
[CI/Linting]   ✅  Success - Install yarn
[CI/Linting] ⭐  Run Install dependencies
[CI/Linting]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/1-composite-2.sh] user= workdir=
yarn install v1.22.19
...
Done in 176.16s.
[CI/Linting]   ✅  Success - Install dependencies
[CI/Linting]   ✅  Success - Setup Node
[CI/Linting] ⭐  Run Lint
[CI/Linting]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/2] user= workdir=
yarn run v1.22.19
$ next lint
| ...
| ✔ No ESLint warnings or errors
Done in 14.57s.
[CI/Linting]   ✅  Success - Lint
[CI/Linting] ⭐  Run Typecheck
[CI/Linting]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/3] user= workdir=
yarn run v1.22.19
$ tsc --noEmit
Done in 33.74s.
[CI/Linting]   ✅  Success - Typecheck
```

---

and tests:

```shell
$ act --container-architecture linux/amd64 --job unit-tests
...
[CI/Tests]   ✅  Success - actions/setup-node@v3
[CI/Tests] ⭐  Run Install yarn
[CI/Tests]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/1-composite-1.sh] user= workdir=
| found 0 vulnerabilities
[CI/Tests]   ✅  Success - Install yarn
[CI/Tests] ⭐  Run Install dependencies
[CI/Tests]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/1-composite-2.sh] user= workdir=
yarn install v1.22.19
...
Done in 177.68s.
[CI/Tests]   ✅  Success - Install dependencies
[CI/Tests]   ✅  Success - Setup Node
[CI/Tests] ⭐  Run Run tests
[CI/Tests]   🐳  docker exec cmd=[bash --noprofile --norc -e -o pipefail /var/run/act/workflow/2] user= workdir=
yarn run v1.22.19
$ jest
|  PASS  packages/core/connector/index.test.ts (12.451 s)
|  ...
|  PASS  components/function/index.test.tsx (12.059 s)
| Test Suites: 24 passed, 24 total
| Tests:       168 passed, 168 total
| Snapshots:   0 total
| Time:        72.839 s
| Ran all test suites.
Done in 80.01s.
[CI/Tests]   ✅  Success - Run tests
```

**Comment by unholypanda** [comment]
Thanks for opening this. I'm not sure why the checks didn't run here–org permissions should just require that I approve for first-time contributors to the project, but I don't see the option here. I took a slightly different approach with this in #18 if you want to take a look (sorry for stealing this out from under you).

Also thanks for the tip on [act](https://github.com/nektos/act) that helped a lot with local development on these actions.

**Comment by unholypanda** [comment]
Closing this since github actions were added in #18

**Pull request closed without merging**